shopinvader_api_quotation/dependencies.py

```
@@ -0,0 +1,140 @@
+# Copyright 2025 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from typing import Annotated
+
+from fastapi import Depends
+
+from odoo import Command, models
+from odoo.api import Environment
+from odoo.exceptions import MissingError
+
+from odoo.addons.base.models.res_partner import Partner as ResPartner
+from odoo.addons.fastapi.dependencies import (
+    authenticated_partner,
+    authenticated_partner_env,
+)
+from odoo.addons.sale.models.sale_order import SaleOrder
+
+from .schemas.sale import (
+    QuotationAddLineRequest,
+    QuotationCreateRequest,
+    QuotationDeleteLineRequest,
+    QuotationLines,
+    QuotationUpdateLineRequest,
+)
+
+
+class ShopinvaderApiQuotationRouterHelper(models.AbstractModel):
+    _name = "shopinvader_api_quotation.router.helper"
+    _description = "Shopinvader api quotations router helper"
+    _inherit = "shopinvader_api_sale.sales_router.helper"
+
+    def _get_domain_adapter(self):
+        return [
+            ("partner_id", "=", self.partner.id),
+            ("typology", "=", "quote"),
+        ]
+
+    def _process_confirm_quotation(self, quotation, data):
+        """Process the quotation confirmation
+        Can be inherited if you expect specific params
+        for confirming a quotation"""
+        return quotation.action_confirm_quotation()
+
+    def _confirm(self, quotation_id, data):
+        quotation = self._get(quotation_id)
+        self._process_confirm_quotation(quotation, data)
+        return quotation
+
+    def _update(self, quotation_id, data):
+        quotation = self._get(quotation_id)
+        quotation.write(data.to_sale_order_vals())
```

If the lines fields is provided into the update method, we should consider that at the end of the process, our quotation should only contains the same number of lines as the one provided...
* If the line has an id -> update existing line (raise if not found)
* If the line has no id -> create a new one
* All existing lines not found in lines should be removed

To know which fields are part of the initial json you can use the method `values = self.model_dump(exclude_unset=True)` values is a dict with only fields provided into the json.